src/renderer/renderer.ts

```diff
@@ -3124,9 +3061,26 @@ export class SDFGRenderer {
 
 
 function calculateNodeSize(
-    _sdfg_state: any, node: any, ctx: CanvasRenderingContext2D
+    sdfg: JsonSDFG, node: any, ctx: CanvasRenderingContext2D
 ): { width: number, height: number } {
-    const labelsize = ctx.measureText(node.label).width;
+    let label;
+    switch (node.type) {
```

Comment:
I wonder, why can't we call the label function instead of copying this "get property" method here?

Reply:
Yeah I was a bit disappointed initially about this too - the issue is that the nodes we're working with at this point are the `JsonSDFG` versions of nodes. However, the label function is only applicable to the proper `SDFGElement` subclass objects (i.e., the `AccessNode` objects). The code could be jossled around to construct this object a priori and then update the size again later, but I believe this would reduce legibility quite a bit. Mostly because certain objects (e.g., `NestedSDFGNode`) require layouting of subgraphs to be completed before their object is constructed.